eng/bomgenerator/src/main/java/com/azure/tools/bomgenerator/DependencyAnalyzer.java

```diff
@@ -0,0 +1,238 @@
+    }
+
+    public boolean validate() {
+        analyze();
+        return nameToVersionToChildrenDependencyTree.keySet().stream().anyMatch(key -> nameToVersionToChildrenDependencyTree.get(key).size() > 1);
+    }
+
+    private void analyze() {
+        pickCoreDependencyRoots();
+        resolveTree();
+        resolveConflicts();
+        filterConflicts();
+    }
+
+    private static List<BomDependency> getDependencies(MavenDependency dependency) {
+        try {
+
+            MavenResolvedArtifact mavenResolvedArtifact = null;
+
+            mavenResolvedArtifact = getMavenResolver()
+                .addDependency(dependency)
+                .resolve()
+                .withoutTransitivity()
+                .asSingleResolvedArtifact();
+
+            return Arrays.stream(mavenResolvedArtifact.getDependencies()).map(mavenDependency ->
+                new BomDependency(mavenDependency.getCoordinate().getGroupId(),
+                    mavenDependency.getCoordinate().getArtifactId(),
+                    mavenDependency.getCoordinate().getVersion(),
+                    mavenDependency.getScope())).collect(Collectors.toList());
+        }
+        catch (Exception ex) {
+            logger.error(ex.toString());
+        }
+
+        return new ArrayList<>();
+    }
+
+    private void resolveTree() {
+        for (MavenDependency gaLibrary : inputDependencies) {
+            try {
+
+                BomDependency parentDependency = new BomDependency(gaLibrary.getGroupId(), gaLibrary.getArtifactId(), gaLibrary.getVersion());
+                addDependencyToDependencyTree(parentDependency, null, nameToVersionToChildrenDependencyTree);
+
+                List<BomDependency> dependencies = getDependencies(gaLibrary);
+                for (BomDependency dependency : dependencies) {
+                    if (dependency.getScope() == ScopeType.TEST) {
+                        continue;
+                    }
+                    if(RESOLVED_EXCLUSION_LIST.contains(dependency.getArtifactId())) {
+                        continue;
+                    }
+
+                    BomDependency childDependency = new BomDependency(dependency.getGroupId(), dependency.getArtifactId(), dependency.getVersion());
+                    addDependencyToDependencyTree(childDependency, parentDependency, nameToVersionToChildrenDependencyTree);
+                }
+            }
+            catch(Exception ex) {
+                System.out.println(ex);
+            }
+        }
+    }
+
+    private static MavenResolverSystemBase<PomEquippedResolveStage, PomlessResolveStage, MavenStrategyStage, MavenFormatStage> getMavenResolver() {
+        return Maven.configureResolver().withMavenCentralRepo(true);
+    }
+
+    private static void addDependencyToDependencyTree(BomDependency dependency, BomDependency parentDependency, Map<BomDependencyNoVersion, HashMap<String, Collection<BomDependency>>> dependencyTree) {
+        if (!dependencyTree.containsKey(dependency)) {
```

Comment:
This can use `computeIfAbsent` to be easier to read. Additionally it can be chained to get the Map from this Map, removing the get below.